Dockerfile

```diff
@@ -1,10 +1,27 @@
 # build sqld
-FROM rust:slim-bullseye AS builder
+FROM rust:slim-bullseye AS chef
 RUN apt update \
     && apt install -y libclang-dev clang \
         build-essential tcl protobuf-compiler file \
         libssl-dev pkg-config git\
-    && apt clean
+    && apt clean \
+    && cargo install cargo-chef
+# We need to install and set as default the toolchain specified in rust-toolchain.toml
+# Otherwise cargo-chef will build dependencies using wrong toolchain
+# This also prevents planner and builder steps from installing the toolchain over and over again
+COPY rust-toolchain.toml rust-toolchain.toml
+RUN cat rust-toolchain.toml | grep "channel" | awk '{print $3}' | sed 's/\"//g' > toolchain.txt \
+    && rustup update $(cat toolchain.txt) \
+    && rustup default $(cat toolchain.txt) \
+    && rm toolchain.txt rust-toolchain.toml
```

Comment:
Is this still needed? If the toolchain file is in the root of the cargo workspace it should pick it up?

Reply:
It is entirely possible. Let me check and get back to you :)

Reply:
Actually even if rust-toolchain.toml is picked up. Not running rustup update here will cause the toolchain to be installed on each step instead of installing it once so this way it's faster and cached better.